libs/react/src/lib/react.tsx

```diff
@@ -200,7 +200,13 @@ export function useResource<T, R>(
   projectFn: (value: T) => R
 ) {
   if (!projectFn(subject.getValue())) {
-    throw subject.pipe(map(projectFn), take(1)).toPromise();
+    throw subject
+      .pipe(
+        map(projectFn),
+        filter((value: unknown) => !!value),
```

Comment:
Is there a reason not to just `filter(projectFn)` ? (might be a type issue)
I think that the thrown promise doesn't have to be resolved to a particular value, just to be resolved at the right time.

Reply:
Yeah, it would be nice (and I admittedly didn't consider that)...

However, it looks like the typings for `filter` are very strict about the callback it is passed, which must return a boolean:

![Screen Shot 2020-09-03 at 6 52 12 AM](https://user-images.githubusercontent.com/4021306/92123986-13c72000-edb2-11ea-8f2f-de8cec4b468f.png)


Usually if I have to supply generic arguments, I feel like it's a smell, but I tried this anyway & didn't seem like it was going in any direction where it would make the code better:

![Screen Shot 2020-09-03 at 6 51 55 AM](https://user-images.githubusercontent.com/4021306/92123990-1590e380-edb2-11ea-88e5-2efe400f5f6b.png)

Let me know if there's another option to improve it I overlooked!

Reply:
I guess maybe one way would be `filter(value => !!projectionFn(value))`, but IMO separate map/filter is more readable.

I did try it out just now, though:
![Screen Shot 2020-09-03 at 7 09 11 AM](https://user-images.githubusercontent.com/4021306/92125960-76b9b680-edb4-11ea-8d04-9d19abb16e64.png)


> the resolve value of the promise doesn't matter
I updated the code this morning, it does matter now FYI (I want the hook to return a "resource" object, so the resource can be set into React state in a transition, as shown here https://reactjs.org/docs/concurrent-mode-patterns.html#splitting-high-and-low-priority-state):

![Screen Shot 2020-09-03 at 7 09 30 AM](https://user-images.githubusercontent.com/4021306/92125953-75888980-edb4-11ea-8938-ed5be69e6703.png)